### ISSUE TITLE
Progress ring blinks on the first animation.

### ISSUE THREAD

**Issue by EugeneEl** [luispadron/UICircularProgressRing#159]
## Bug Report

Progress ring blinks on the first animation.

#### Version

4.1.0

#### Overview of what you tried to do

I've tried to reset progress, decrease timing. 

#### What is the expected outcome?

Circle fill animation should be smooth on the first time.

#### What is the actual outcome?

There is a small blink with the next value. than progress is set to 0 and animation starts.

```swift
Swift 4.2
```
#### Have you tried fixing this yourself?

No.

**Comment by luispadron** [comment]
Can you post some sample code or gif or some sort of example project where this is replicated? I haven't experienced this issue before.

**Comment by EugeneEl** [comment]
https://www.dropbox.com/s/7p7ygh1vtwtoonx/oolc4-dot17.mov?dl=0
I've recorder a video.

Note:
1. On device it is reproducible much easier. 
2. Sometimes it blinks when the first screen is displayed.
3. It is UIPageViewController instance (swipe left/right)
So reuse is not supposed to be a reason in this case.

**Comment by luispadron** [comment]
Yeah I'm not really seeing anything. This isn't anything I can fix if there was an issue. The animation is done using CoreGraphics which is an Apple API if the device or simulator cannot process the animation smoothly or if it they're both under load and thus have stutters in animation, this isn't something I can fix or a bug with the library.

I'm going to close this issue since it's not a "bug" I can fix.

**Comment by EugeneEl** [comment]
It is reproducible only first time. Sometimes. 
Next progress value updates works ok.
Any suggestions what I can try to add in order to fix this?

**Comment by luispadron** [comment]
This doesn't seem like anything that can be fixed. This seems like possibly a hardware issue, either device or computer running the application is slow or under load thus causing animation lag. Not something that can be fixed in this library.

**Comment by EugeneEl** [comment]
I have some suggestions that it can be related to UIPageViewController which has quite awful API and implementation. 
Because on standalone screen it usually works ok.
I don't think it is simulator issue, because simulator uses laptop hardware (as it is not emulator)

**Comment by luispadron** [comment]
It may be that, but again that’s apple api and nothing I can fix. Maybe file a radar for apple.

**Comment by EugeneEl** [comment]
Note.
I tested some other progress ring libraries, but still have glitching on the first time.
Actually all of them use the same CoreAnimation and CoreGraphics synergy, common drawing logic.

Also I made an attempt to replace UIPageViewController with UICollectionView+controllers in cells.
But it did't resolve an issue.

What I found out is that I'm making API request on viewDidLoad().
And if UI configuration with response data occurs too fast  (before layout precess finished) - it is very reproducible.

So I decided 
1 make preloadData on viewDidAppear once.
2 add a small delay before request.  

With this hacks it works quite better.

Seems to be very internal CoreAnimation issue during layout process.
